code/components/gta-game-five/src/Timecycle.cpp

```diff
@@ -0,0 +1,767 @@
+#include <StdInc.h>
+#include <Hooking.h>
+#include <Hooking.Stubs.h>
+#include <Timecycle.h>
+#include <GameInit.h>
+
+static TimecycleScriptData* g_timeCycleScriptData;
+
+rage::tcManager* TimecycleManager::GetGameManager()
+{
+	assert(rage::tcManager::ms_Instance);
+	return rage::tcManager::ms_Instance;
+};
+
+TimecycleScriptData* TimecycleManager::GetScriptData()
+{
+	return g_timeCycleScriptData;
+};
+
+std::string& TimecycleManager::GetTimecycleName(uint32_t hash)
+{
+	if (auto& it = m_originalNames.find(hash); it != m_originalNames.end())
+	{
+		return it->second;
+	}
+
+	if (auto& it = m_customNames.find(hash); it != m_customNames.end())
+	{
+		return it->second;
+	}
+
+	assert(false && va("Failed to find timecycle name for %X", hash));
```

Comment:
`assert` won't format this- it stringifies the entire argument. `FatalError` would work equally here except it may run into issues if it's not annotated as no-return.

(similarly, returning at least _some_ valid string here might prevent bad compiler optimizations)